src/main/java/UI/GameBoard.java

```diff
@@ -96,33 +101,43 @@ public void createPlayers(JLabel bg) {
     }
 
     public void updatePlayer1Card(String cardStr) {
-        JLabel card = createCardLabel(80, 460, 85, 120, cardStr);
-        gameLabel.add(card);
+        if (cardStr.equals("D")) {
+            updateDrawSymbol(this.player1Card, 80, 460, 85, 120);
+        } else {
+            updateCardLabel(this.player1Card, 80, 460, 85, 120, cardStr);
+        }
```

Comment:
Lines 104 - 108 are repeated four times. Perhaps you could make a helper method that holds it?